### PR TITLE
ADD .test covering date filtering subscription cases

### PR DESCRIPTION
**Pull request by fgalan**
@kzangeli 
@crbrox

### CODE REVIEW

File: src/lib/mongoBackend/MongoCommonUpdate.cpp [hunk 1]
```diff
@@ -2868,6 +2868,11 @@ static void updateEntity
   AttributeList emptyAttrL;
   ContextElementResponse* notifyCerP = new ContextElementResponse(r, emptyAttrL);
 
+  // The hasField() check is needed as the entity could has been created with very old Orion version not
```
Comment (kzangeli) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: has => have

File: test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+        "attributes": [ ]
+    },
+    "expires": "2050-04-05T14:00:00.00Z"
+}
+'
+orionCurl --url /v2/subscriptions --payload "$payload" --json
+echo
+echo
+sleep 1s
+
```
Comment (kzangeli) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: You might need a little more margin to make this test work also under valgrind

Comment (fgalan) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: You have more experience on that... any suggestion in on how to re-time the test to make it work under valgrind?

File: src/lib/mongoBackend/MongoCommonUpdate.cpp (hunk 1 above)
Comment (fgalan) [inline src/lib/mongoBackend/MongoCommonUpdate.cpp]: Fixed in 795e21d396d1ea3b0c6c2599cd843a37b53eeb2e

File: test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: I would simply test it under valgrind.  (-filter filter_datecreated_sub.test)
Difficult to guess ...

Comment (fgalan) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: I have tested:

```
fermin@neodeb:~/src/fiware-orion/test/valgrind$ ./valgrindTestSuite.sh -filter filter_datecreated_sub.test
jue feb 25 13:47:48 CET 2016
./valgrindTestSuite.sh: línea 218: cd: test/valgrind: No existe el fichero o el directorio
Test 001/1: 0876_entity_dates/filter_datecreated_sub ................................................................................................ OK (76.75 seconds)
Total test time: 76.85 seconds (1 minutes)
Great, all valgrind tests ran without any memory leakage

fermin@neodeb:~/src/fiware-orion/test/valgrind$ ./valgrindTestSuite.sh -filter filter_datemodified_sub.test
jue feb 25 13:50:12 CET 2016
./valgrindTestSuite.sh: línea 218: cd: test/valgrind: No existe el fichero o el directorio
Test 001/1: 0876_entity_dates/filter_datemodified_sub ............................................................................................... OK (48.09 seconds)
Total test time: 48.16 seconds (0 minutes)
Great, all valgrind tests ran without any memory leakage
```

So I think everything is ok.

Thus, NTC?

**Comment by crbrox** [comment]
LGTM

File: test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: NTC

Comment (kzangeli) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: Wait ...
línea 218: cd: test/valgrind: No existe el fichero o el directorio  ? :-)

Comment (fgalan) [inline test/functionalTest/cases/0876_entity_dates/filter_datecreated_sub.test]: That's correct, but I understand that is a bug in the valgrindTestSutie.sh script, not related with the PR.

**Comment by kzangeli** [comment]
LGTM

**Pull request merged**